src/common/selectors/index.js

```diff
@@ -0,0 +1,20 @@
+import { createSelector } from 'reselect'
+
+import { parseGitHubRepoUrl } from '../helpers/github-url.js';
+
+const getSnaps = state => state.snaps;
+const getNames = state => state.registerName;
+
+export const hasNamedSnaps = createSelector(
+  [getSnaps, getNames],
+  (snaps, names) => {
+    return snaps.snaps.some((snap) => {
+      if (typeof(snap.store_name) === 'string') return true;
+
+      const { fullName } = parseGitHubRepoUrl(snap.git_repository_url);
```

Comment:
I think this is (or at least should be) unnecessary.
Snaps reducer already listens to register action and updates `store_name` directly in the store, so we don't even need to re-fetch snaps from LP to get this change: https://github.com/canonical-ols/build.snapcraft.io/blob/master/src/common/reducers/snaps.js#L61-L65

So if the name was registered by user in BSI, it should immediately be updated in the stored snap.